aopy/analysis.py

```diff
@@ -510,7 +512,303 @@ def get_unit_spiking_mean_variance(spiking_data):
+        vel (nt): time series for cursor velocity (magnitude)
+        num_samples_before (int): number of samples to extract before each saccade
+        num_samples_after (int): number of samples to extract after each saccade
+        
+    Returns:
+        tuple: tuple containing:
+                | **cursor_dists (dict):** value is list of timeseries (one per saccade) for distance of cursor from target
+                | **cursor_vels (dict):** value is list of timeseries (one per saccade) for velocity of cursor
+                | **dist_slope (dict):** value is change in distance of cursor from target in period, normalized by length of period
+                | **dist_avg (dict):** value is average distance of cursor from target in period
+                | **vel_slope (dict):** value is change in velocity of cursor in period, normalized by length of period
+                | **vel_avg (dict):** value is average velocity of cursor in period
+    
+    Note: All dictionary keys are ['before', 'during', 'after']
+    '''
+    
+    periods = ['before', 'during', 'after']  # relative to saccade
+    
+    cursor_dists = {}
+    cursor_vels = {}
+    dist_slope = {}
+    dist_avg = {}
+    vel_slope = {}
+    vel_avg = {}
+    
+    for p in periods:
+        cursor_dists[p] = []
+        cursor_vels[p] = []
+        dist_slope[p] = []
+        dist_avg[p] = []
+        vel_slope[p] = []
+        vel_avg[p] = []
+        
+    win_start = {}; win_end = {}
+    for idx_sac in range(len(sac_start_times)):
+        win_start['before'] = sac_start_times[idx_sac]-num_samples_before
+        win_end['before'] = sac_start_times[idx_sac]+1  # last sample includes saccade
+
+        win_start['after'] = sac_end_times[idx_sac]  # first sample includes saccade
+        win_end['after'] = sac_end_times[idx_sac]+num_samples_after 
+
+        win_start['during'] = sac_start_times[idx_sac]  # first and last samples include saccade
+        win_end['during'] = sac_end_times[idx_sac]+1
+
+        for p in periods:
+            cursor_dists[p].append(dist[win_start[p]:win_end[p]])  # slicing captures correct window even if start/end are out of bounds
+            cursor_vels[p].append(vel[win_start[p]:win_end[p]])
+
+            dist_avg[p].append(np.mean(dist[win_start[p]:win_end[p]]))
+            vel_avg[p].append(np.mean(vel[win_start[p]:win_end[p]]))
+
+            win_start[p] = np.max([0, win_start[p]])  # shrink windows to within bounds
+            win_end[p] = np.min([len(dist)-1, win_end[p]])
+            
+            if win_start[p] != win_end[p]:
+                dist_slope[p].append((dist[win_end[p]]-dist[win_start[p]])/(win_end[p]-win_start[p]))
+                vel_slope[p].append((vel[win_end[p]]-vel[win_start[p]])/(win_end[p]-win_start[p]))
+                
+    return cursor_dists, cursor_vels, dist_slope, dist_avg, vel_slope, vel_avg
+
+def compute_directional_error(pos, target, use_initial_direction = False):
```

Comment:
There’s an existing function get_inst_target_dir which I think does the same thing as this. Maybe take a look before writing any tests for this one.